lib/CMakeLists.txt

```diff
@@ -6,41 +6,29 @@ project(tree-sitter
         HOMEPAGE_URL "https://tree-sitter.github.io/tree-sitter/"
         LANGUAGES C)
 
-if(NOT MSVC)
-  set(CMAKE_C_FLAGS "-O3 -Wall -Wextra -Wshadow -Wno-unused-parameter -pedantic")
-endif(NOT MSVC)
-
 option(BUILD_SHARED_LIBS "Build using shared libraries" ON)
 option(TREE_SITTER_FEATURE_WASM "Enable the Wasm feature" OFF)
 
-file(GLOB TS_SOURCE_FILES src/*.c)
-list(REMOVE_ITEM TS_SOURCE_FILES "${PROJECT_SOURCE_DIR}/src/lib.c")
-
-add_library(tree-sitter ${TS_SOURCE_FILES})
+add_library(tree-sitter src/lib.c)
 
 target_include_directories(tree-sitter PRIVATE src src/wasm include)
+if(NOT MSVC)
+  target_compile_options(tree-sitter PRIVATE -O3 -Wall -Wextra -Wshadow -Wno-unused-parameter -pedantic)
+endif()
 
 if(TREE_SITTER_FEATURE_WASM)
-  if(NOT DEFINED CACHE{WASMTIME_INCLUDE_DIR})
-    message(CHECK_START "Looking for wasmtime headers")
-    find_path(WASMTIME_INCLUDE_DIR wasmtime.h
-              PATHS ENV DEP_WASMTIME_C_API_INCLUDE
-              REQUIRED)
-    message(CHECK_PASS "found")
-  endif(NOT DEFINED CACHE{WASMTIME_INCLUDE_DIR})
-
-  if(NOT DEFINED CACHE{WASMTIME_LIBRARY})
-    message(CHECK_START "Looking for wasmtime library")
-    find_library(WASMTIME_LIBRARY wasmtime
-                 REQUIRED)
-    message(CHECK_PASS "found")
-  endif(NOT DEFINED CACHE{WASMTIME_LIBRARY})
+  find_path(WASMTIME_INCLUDE_DIR wasmtime.h
+            PATHS ENV DEP_WASMTIME_C_API_INCLUDE
+            REQUIRED)
+
+  find_library(WASMTIME_LIBRARY wasmtime
+               REQUIRED)
```

Comment:
We could use https://github.com/neovim/neovim/blob/master/cmake/FindWasmtime.cmake if we want to have a minimum/exact wasmtime version check. If I'm not mistaked then neovim and tree-sitter should require the same wasmtime versions, but pinging @clason for confirmation.

Reply:
Well, obviously the intersection of valid ranges cannot be empty... But it is a very good idea to ensure that a compatible version is found since distros will 1. require using system wasmtime and 2. yolo upgrade every chance they get (which is once a month on the 20th).

From what it looks like, breaking changes are rare even in major version updates (unless we trigger them...) but you'll never know. If Neovim goes full wasm, then _not_ being able to rely on builds having a known-good version would be a dealbreaker.

Reply:
Makes sense. Cmake requires using `find_package`/`FindWasmtime.cmake` but that makes the code somewhat more complex as seen here. If it's not strictly needed now but later then we can revisit this later on when we need this.